src/parser.ts

```diff
@@ -3115,14 +3119,12 @@ class Parser {
             this.throwUnexpectedToken(this.lookahead);
         }
 
-        let property = this.finalize(node, new Node.Property(kind, key, computed, value, method, shorthand));
-        property.static = isStatic;
-        if (property.kind === 'init') {
-            property.kind = 'method';
+        if (kind === 'init') {
+            kind = 'method';
         }
         if (!isStatic) {
-            if (!property.computed && (property.key.name || property.key.value.toString()) === 'constructor') {
-                if (property.kind !== 'method' || !property.method || property.value.generator) {
+            if (!computed && this.isPropertyKey(key, 'constructor')) {
```

Comment:
is there any test you can add that checks the `(key.type === Syntax.Literal && key.value === value)` part of `this.isProperty` for this condition?